Packs/AppSentinelsAi/Integrations/AppSentinelsAi/AppSentinelsAi.py

```diff
@@ -0,0 +1,347 @@
+            use_proxy: True if the proxy server needs to be used, False otherwise.
+        """
+
+        super().__init__(base_url=base_url, verify=verify, proxy=use_proxy)
+        self._headers = {
+            "accept": "application/json",
+            "apikey": api_key,
+            "x-user-key": user_key,
+            "Content-Type": "application/json",
+        }
+        self.organization = organization
+        self.base_request_body = base_request_body
+        self.base_request_params = base_request_params
+        self.api_key = api_key
+        self.user_key = user_key
+
+    def get_events_request(self, params_update: dict, body_update: dict) -> dict:
+        """
+        Retrieve the detections from AppSentinels.ai  API.
+
+        Args:
+            params_update (dict): The param update to add to the base params.
+            body_update (dict): The body update to add to the base body.
+
+        """
+        url_suffix = f"/api/v1/{self.organization}/audit-logs"
+        params = self.base_request_params.copy()
+        params.update(params_update)
+        body = self.base_request_body.copy()
+        body.update(body_update)
+        return self._http_request(
+            "POST", url_suffix=url_suffix, headers=self._headers, json_data=body, params=params, resp_type="json"
+        )
+
+
+""" HELPER FUNCTIONS """
+
+
+def remove_first_run_params(params: dict[str, Any]) -> None:
+    """
+    Remove the "First Run" items from the param dictionary.
+
+    Args:
+        params (Dict[str, Any]): Integration parameters.
+
+    """
+    for key in ("from_date", "to_date"):
+        params.pop(key, None)
+
+
+def fetch_events_list(client: Client, last_run: Dict, fetch_limit: int | None, use_last_run_as_body: bool) -> List[Dict]:
+    """
+    Fetches events from the AppSentinels.ai API, handling pagination and last_run.
+
+    Args:
+        client (Client): The client object for interacting with the AppSentinels.ai API.
+        last_run (Dict): A dictionary containing the last processed event ID.
+        fetch_limit (Optional[int]): The maximum number of events to fetch.
+        use_last_run_as_body (bool): Flag that sign do we use the last-run as params for the API call
+
+    Returns:
+        List[Dict]: A list of fetched events.
+    """
+    events: List[Dict] = []
+    current_pagination: int = 0
+    params: Dict[str, Any] = {}  # Initialize params
+    body: Dict[str, Any] = {}  # Initialize body
+
+    # Determine the fetch params
+    if use_last_run_as_body:
+        body.update(last_run)
+        demisto.debug(f"AppSentinels.ai Fetching with {body=}")
+    elif "last_log_id" not in last_run:
+        # Initial fetch: from one minute ago to now
+        current_time = get_current_time()
+        start_time = (current_time - timedelta(minutes=1)).strftime(DATE_FORMAT)
+        end_time = current_time.strftime(DATE_FORMAT)
+        body["from_date"] = start_time
```

Comment:
You can just update the body directly:
        body.update({
            "from_date": (now - timedelta(minutes=1)).strftime(DATE_FORMAT),
            "to_date": now.strftime(DATE_FORMAT),
        })